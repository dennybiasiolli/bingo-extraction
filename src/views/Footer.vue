<template>
  <v-footer>
    Remaining numbers: {{ remainingNumbers }}
    <v-spacer />
    Latests 5 extracted numbers: [{{ reversedExtractedNumbers.slice(0, 5).join(', ') }}]
  </v-footer>
</template>

<script>
import { mapState, mapGetters } from 'vuex';

export default {
  computed: {
    ...mapState({
      remainingNumbers: (state) => state.availableNumbers.length,
    }),
    ...mapGetters(['reversedExtractedNumbers']),
  },
};
</script>
