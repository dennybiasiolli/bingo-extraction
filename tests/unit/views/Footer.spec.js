import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Footer from '@/views/Footer.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Footer.vue', () => {
  let state; let getters; let store;
  beforeEach(() => {
    state = {
      availableNumbers: [...Array(15).keys()].map((i) => i + 1),
      extractedNumbers: [1, 2, 3, 4, 5, 6, 7, 8, 9],
    };
    getters = {
      reversedExtractedNumbers: (s) => [...s.extractedNumbers].sort((a, b) => a - b),
    };
    store = new Vuex.Store({ state, getters });
  });

  it('renders as expected', () => {
    const wrapper = shallowMount(Footer, {
      store,
      localVue,
      stubs: ['v-footer', 'v-spacer'],
    });
    expect(wrapper).toMatchSnapshot();
  });
});
