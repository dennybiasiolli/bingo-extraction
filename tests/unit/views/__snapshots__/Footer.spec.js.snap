// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.vue renders as expected 1`] = `
<v-footer-stub>
  Remaining numbers: 15
  <v-spacer-stub></v-spacer-stub>
  Latests 5 extracted numbers: [1, 2, 3, 4, 5]
</v-footer-stub>
`;
